### PR TITLE
New function: script.solveOutput

### PR DESCRIPTION
**Pull request by afk11**
This function builds on `classifyOutput` and the various `decode` functions. It returns an object with the following properties based on the script. 
  - `type` the type of script
  - `script` we include the script itself. 
  - `solvedBy` - the result of calling `decode` for that type of output script
  - `requiredSigs` is null except for P2PKH/P2PK/multisig. 
  - <strike>`canSign` indicates whether the script is signable (defined later). </strike>
 
<strike>It also adds two constants: `SIGNABLE_SCRIPTS`, ie, the ones we support signing (p2pk/p2pkh/multisig). `P2SH_SCRIPTS` is SIGNABLE_SCRIPTS and also P2PKH & P2WSH.</strike>

```js
{
  type: "pubkey",                  // any of the bscript.types
  script: "4104ababababab......ac" // as is
  solvedBy: "04ababababab..",    // ** this is the output of the respective output.decode() function
  requiredSigs: 1                  // set for P2PK, P2PKH, Multisig, P2WPKH
}
```
This is useful when determining if there is enough information to sign, and guides the approach outlined in https://github.com/bitcoinjs/bitcoinjs-lib/issues/695

### CODE REVIEW

File: src/templates/index.js [hunk 1]
```diff
@@ -54,16 +57,69 @@ function classifyWitness (script, allowIncomplete) {
...
+  }
+
+  return {
+    type: outputType,
+    script: scriptCode,
+    solvedBy: solvedBy,
+    canSign: SIGNABLE_SCRIPTS.indexOf(outputType) !== -1,
+    requiredSigs: requiredSigs
+  }
+}
```
Comment (dcousens) [inline src/templates/index.js]: `scripts` has been so far purely about templates... this feels like the wrong place

**Comment by afk11** [comment]
Anything else required for this?

**Comment by dcousens** [comment]
Could you add an integration test to show-case how it might be used?
Or show how it might be more idiomatic then doing it yourself?

**Comment by afk11** [comment]
Yeah, but it captures basically all the state you are deducing with, and gives you something more useful to save as state than `prevOutScript` `redeemScript` `witnessScript` since you probably have to work with the same material again. (signing, then later you need to classify to determine where things go)

**Comment by dcousens** [comment]
@afk11 did you see my comments on slack?

**Pull request closed without merging**